test/controllers/subjects_controller_test.exs

```diff
@@ -0,0 +1,158 @@
+defmodule ActiveMonitoring.SubjectsControllerTest do
+  use ActiveMonitoring.ConnCase
+  use ExUnit.Case
+  import ActiveMonitoring.Factory
+
+  alias ActiveMonitoring.{Repo}
+
+  setup %{conn: conn} do
+    user = build(:user, email: "test@example.com") |> Repo.insert!
+    campaign = build(:campaign, user: user) |> Repo.insert!
+
+    other_user = build(:user, email: "other@example.com") |> Repo.insert!
+    other_campaign = build(:campaign, user: other_user) |> Repo.insert!
+
+    other_subject = build(:subject, campaign: other_campaign) |> Repo.insert!
+
+    {:ok, conn: conn, user: user, campaign: campaign, other_campaign: other_campaign, other_subject: other_subject}
+  end
+
+  defp with_logged_in_user %{conn: conn, user: user} do
+    conn = conn |> assign(:current_user, user)
+    [conn: conn]
+  end
+
+  defp with_campaign_subjects %{campaign: campaign} do
+    subject = build(:subject, campaign: campaign) |> Repo.insert!
+    for _ <- 1..3, do: build(:subject, campaign: campaign) |> Repo.insert!
+    [subject: subject]
+  end
+
+  defp with_many_subjects %{campaign: campaign} do
+    for _ <- 1..100, do: build(:subject, campaign: campaign) |> Repo.insert!
+    :ok
+  end
+
+  describe "campaign without subjects" do
+    setup [:with_logged_in_user]
+
+    test "shows an empty index", %{conn: conn, campaign: campaign} do
+      response = conn |> get(campaigns_subjects_path(conn, :index, campaign)) |> json_response(200)
+      assert response["data"]["subjects"] == []
+      assert response["meta"]["count"] == 0
+    end
+  end
+
+  describe "campaign with subjects" do
+    setup [:with_logged_in_user, :with_campaign_subjects]
+
+    test "lists all subjects", %{conn: conn, campaign: campaign, subject: subject} do
+      assert subject.registration_identifier != nil
```

Comment:
Why do we need to make this assertion?

Reply:
Wanted to make sure the subject actually _has_ a registration identifier, because the test _could_ pass even if the builder didn't set this up. But, on second thought, if we **care** about the registration identifier, we should explicitly set it on the fixture. 👍

Reply:
Yeah, if it's optional, it should be explicitly set on the fixture. If it's mandatory, there should be a test somewhere ensuring that "when I create a subject, it won't save unless it has registration_identifier".